jrnl/plugins/pelican_markdown_exporter.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
```

Comment:
Would you mind splitting adding a new exporter into a separate PR? I think this PR has good fixes for the YAML exporter, but I'd like to be able to talk about adding a new plugin separately from these fixes.

Reply:
The whole point is this isn't adding a new exporter, but rather restoring the YAML exporter that existed in version 2.6 and before. The "pure" YAML is the new plugin.

Reply:
I sense some frustration in your reply (am I reading that right?).

I'll give some more context below, but the gist is that I would really prefer to see this split up into two PRs (a bugfix one, and an enhancement one). Partially because it's easier to review and document smaller changes, but also because I want to be sure you get credit in the changelog for the work you've done.

So, for context: I don't consider changing undocumented functions a breaking change.  Since the docs say this is a YAML format, as long as the output is still valid YAML, then it's not a breaking change in my eyes.

This does raise the question about whether these formats should be considered a stable API or not, but that feels like a conversation more related to the new plugin system that's in the works (thanks again for all your work on that). For the time being, though, these plugins aren't/haven't been stable (YAML isn't the only one that's changed output, and some of them flat out don't work under certain conditions).

So, that's all to say that I think the best way to prevent these types of misunderstandings is to make sure that your work is well documented and tested. I try to review the docs when considering changes, so it's much more likely that a breaking change will be caught during code review if it's in the docs (this also helps other contributors since they wouldn't have to dig through old closed issues to understand what a feature is supposed to do).

Anyway, I understand if you are frustrated, but hopefully you can understand where I'm coming from.